README.md

```diff
@@ -47,9 +47,19 @@ provider puppetca {
 resource "puppetca_certificate" "test" {
   name = "0a7842c26ad0.foo.com"
 }
+
+resource "puppetca_certificate" "ec2instance" {
+  name   = "0a7842c26ad1.foo.com"
+  usedby = aws_instance.ec2instance.id
+}
 ```
 
-or to configure the provider with environment variables:
+The first puppetca_certificate resource, test, will remove the certificate if a destroy plan is run.
+The second puppetca_certificate resrouce, ec2instance, will remove the certificate if Terraform destroys the EC2 instance.
+
+The usedby parameter can be populated with a resource parameter the drives the removal of the certificate from the Puppet CA at the desired time.  In the example above, if a Terraform plan has to recreate the EC2 instance, the certificate will be removed when the EC2 instance is destroyed since each EC2 instance is assigned a new instance id.
```

Comment:
```suggestion
The `usedby` parameter can be populated as a resource parameter to drive the removal of the certificate from the Puppet CA at the desired time.  In the example above, if a Terraform plan has to recreate the EC2 instance, the certificate will be removed when the EC2 instance is destroyed since each EC2 instance is assigned a new instance id.
```